pkg/controller/replication/config.go

```diff
@@ -207,7 +208,8 @@ func (r *ReplicationConfig) reconcileUserSql(ctx context.Context, mariadb *maria
 		return fmt.Errorf("error checking if replication user exists: %v", err)
 	}
 	if exists {
-		if err := client.AlterUser(ctx, opts.username, replPassword); err != nil {
+		maxUserConnections := user.Spec.MaxUserConnections
+		if err := client.AlterUser(ctx, opts.username, replPassword, maxUserConnections); err != nil {
```

Comment:
See my previous comment. This should pass the options like we do for `CreateUser`.

Reply:
Let me know if any change is needed. :D
```
if exists {
		if err := client.AlterUser(ctx, opts.username, sqlClient.WithIdentifiedBy(replPassword)); err != nil {
			return fmt.Errorf("error altering replication user: %v", err)
		}
	}
```